src/main/java/io/jenkins/plugins/llvm/LLVMCovReportDocumentConverter.java

```diff
@@ -0,0 +1,255 @@
+package io.jenkins.plugins.llvm;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.jenkins.plugins.coverage.adapter.converter.JSONDocumentConverter;
+import io.jenkins.plugins.coverage.exception.CoverageException;
+import org.apache.commons.lang.StringUtils;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import java.io.File;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class LLVMCovReportDocumentConverter extends JSONDocumentConverter {
+    @Override
+    protected Document convert(JsonNode report, Document document) throws CoverageException {
+        // only support 2.0.0 version now
+        if (!report.get("version").asText().equals("2.0.0")) {
+            throw new CoverageException("Unsupported Json file - version must be 2.0.0");
+        }
+
+        if (!report.get("type").asText().equals("llvm.coverage.json.export")) {
+            throw new CoverageException("Unsupported Json file - type must be llvm.coverage.json.export");
+        }
+
+        Element reportEle = document.createElement("report");
+        reportEle.setAttribute("name", "llvm-cov");
+        document.appendChild(reportEle);
+
+        JsonNode dataArr = report.get("data");
```

Comment:
Is it possible for the report to not have a data node? If so would it be good to verify and throw an explicit exception here?